### ISSUE TITLE
GPU Offloading not Working - Cuda not detected

### ISSUE THREAD

**Issue by Iory1998** [gokayfem/ComfyUI_VLM_nodes#107]
Llama.cpp seems to use CPU instead of the GPU (RTX3090) which makes the process very slow,
No matter the number of GPU layers I set, the model will always be offloaded to the CPU. Also, It's seems that BLAS is activated but not used?
![image](https://github.com/user-attachments/assets/81abd360-fbb2-4255-8830-dee0dee15baa)

**Comment by doggeddalle** [comment]
-_-

**Comment by gokayfem** [comment]
Yeah, even if you install it from llama-cpp-python library with CMAKE ARGS 3090 has this problem, i could not find a solution, i have 3090 too..

**Comment by DavideAlidosi** [comment]
I currently have two installations of ComfyUI, on the older one the GPU acceleration works, on the new one it does not. I did some research and noticed how the defect is not directly on Comfy but on some dependency, find where the real problem is though.
I am also using a 3090, but my experience excludes a HW problem.

**Comment by liammellofarley** [comment]
Managed to install Llama and everything and can run it manually with CUDA but it fails in comfyui to use CUDA.

**Comment by gokayfem** [comment]
The GGUF runtime was rebuilt and now exposes the controls this needs directly: `gpu_layers`, `split_mode`, `main_gpu`, `tensor_split`, and a `flash_attention` mode that safely retries without it when unsupported.

Whether llama.cpp was actually built with CUDA is now inspectable — add the **VLM Runtime Diagnostics** node, which reports the detected backend and the llama.cpp build info rather than leaving you to guess.

The underlying cause was almost always a CPU-only `llama-cpp-python` wheel. Install the accelerator-specific build from the [upstream guide](https://github.com/abetlen/llama-cpp-python#installation).

If this still happens on **3.3.1**, please reopen with the new bug report template — it asks for the `VLM Runtime Diagnostics` node output, which captures the environment detail these older reports were missing.
